x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -569,11 +569,6 @@ private LogicalPlan resolveEnrich(Enrich enrich, List<Attribute> childrenOutput)
                 if (resolved.equals(ua)) {
                     return enrich;
                 }
-                if (resolved.resolved() && resolved.dataType() != KEYWORD) {
```

Comment:
We might want to bring this back for anything we don't support fully, as this provides a much nicer error message than the failure we get later.

Reply:
Why not add it now? How many supported types are there?

Reply:
We did add it back. My comment was from my first commit (actually Nhat's stub commit that removed it), but I brought it back almost directly, and have refined it a lot since then.